test/__snapshots__/index.js.snap

```diff
@@ -67,6 +67,54 @@ const Test = () => <div data-jsx={1535297024}>
   </div>;"
 `;
 
+exports[`works with array children 1`] = `
+"import _JSXStyle from \\"styled-jsx/style\\";
+/* eslint-disable no-unused-vars */
+
+const Test1 = () => [<div data-jsx={465566775}>Foo</div>, <span data-jsx={465566775}>Bar</span>, <_JSXStyle styleId={465566775} css={\\"div[data-jsx=\\\\\\"465566775\\\\\\"]{color:red}\\"} />];
+
+const Test2 = () => [<p data-jsx={2639504339}>Hello World</p>, <_JSXStyle styleId={2639504339} css={\\"p{color:white}\\"} />];
+
+const Test3 = () => [<div data-jsx={2293543665}>
+        <p data-jsx={2293543665}>Hello World</p>
+        <_JSXStyle styleId={2293543665} css={\\"p[data-jsx=\\\\\\"2293543665\\\\\\"]{color:blue}\\"} />
+    </div>, <div>Hello World</div>];
+
+const Test4 = () => [<div data-jsx={1371748502}>
```

Comment:
This case is the one that's giving me the most trouble.

I'm not sure what the desired behavior would be in this situation but, as you can see, if there is a div element with a nested `<style>` component **and** a top-level `<style>` component, the top level component wins.

IMHO, this situation should merge the nested style rules with the top level style rules in each of the nested style components and keep the style rules as-is in the top-level component. 

Any ideas?